src/protocol/malicious.rs

```diff
@@ -115,19 +113,11 @@ impl<F: Field> MaliciousValidatorAccumulator<F> {
 
     /// ## Panics
     /// Will panic if the mutex is poisoned
-    pub fn accumulate_macs(
-        &self,
-        prss: &Arc<IndexedSharedRandomness>,
-        record_id: RecordId,
-        input: &MaliciousReplicated<F>,
-    ) {
+    pub fn accumulate_macs(&self, random_constant: &Replicated<F>, input: &MaliciousReplicated<F>) {
```

Comment:
Can this be reverted now?

Reply:
This interface definitely feels more error prone, since there is only one way to do this correctly. I would prefer the original interface.

Reply:
The function could take a context as an argument instead.  Would that work better?

Reply:
all right you convinced me